packages/veraswap-sdk/src/swap/getSwapAndHyperlaneSweepBridgeTransaction.ts

```diff
@@ -19,44 +16,35 @@ export function getSwapAndHyperlaneSweepBridgeTransaction({
     destinationChain,
     receiver,
     amountIn,
-    amountOutMinimum,
     currencyIn,
-    currencyOut,
-    path,
+    routePlanner,
     permit2PermitParams,
-    hookData = "0x",
 }: {
     universalRouter: Address;
     bridgeAddress: Address;
     bridgePayment: bigint;
     destinationChain: number;
     receiver: Address;
     amountIn: bigint;
-    amountOutMinimum: bigint;
     currencyIn: Address;
-    currencyOut: Address;
-    path: PathKey[];
+    routePlanner: RoutePlanner;
     permit2PermitParams?: [PermitSingle, Hex];
-    hookData?: Hex;
 }) {
-    const routePlanner = new RoutePlanner();
+    const finalRoutePlanner = permit2PermitParams ? new RoutePlanner() : routePlanner;
 
     if (permit2PermitParams) {
-        routePlanner.addCommand(CommandType.PERMIT2_PERMIT, permit2PermitParams);
+        finalRoutePlanner.addCommand(CommandType.PERMIT2_PERMIT, permit2PermitParams);
+        const hexCommands = routePlanner.commands.slice(2); // remove "0x"
+        const commands: CommandType[] = [];
+        for (let i = 0; i < hexCommands.length; i += 2) {
+            commands.push(Number(hexCommands.slice(i, i + 2)));
+        }
+        commands.forEach((commandType, i) => {
+            finalRoutePlanner.addCommand(commandType, [routePlanner.inputs[i]]);
+        });
```

Comment:
1: refactor this in to a function
2: @leovigna pls just return an array of commands and inputs rather than a routerPlanner. would be so much cleaner lol
(Also you might still get nice type checks in the quoting)